# This script takes a .txt file generated by bedtools genomecov -d, and converts it to WIG format
# to be visualized on IGV. The .txt file has 3 columns: (Chromosome) (Nucleotide) (Read count).
# Read counts may have been normalized after the original .txt file was created by BEDtools.
# To run: python3 make_wig_from_nucleotide.py {input_filename.txt} {output_filename.wig}.
# NOTE: Must use python3 because of formatted strings.

from sys import argv

if len(argv) != 3:
	print("Run with 2 arguments: input_filename.txt output_filename.wig")
	exit(0)
else:
	pass
	
script, input_filename, output_filename = argv

current_file = open(input_filename)
write_page = open(output_filename, 'w')

write_page.write("track type=wiggle_0 name=track_label" + '\n')

for line in current_file:
	current_line = line.split('\t')
	chromosome = current_line[0]
	nucleotide = int(current_line[1])
	read_count = float(current_line[2])
	
	if (nucleotide == 1):
		write_page.write(f"fixedStep chrom={chromosome} start=1 step=1\n")
	else:
		pass
		
	write_page.write(f"{read_count}\n")
	
write_page.close()
	
	
